### ISSUE TITLE
Panics on C diff: `Should have changes set in all nodes`

### ISSUE THREAD

**Issue by sdroege** [Wilfred/difftastic#172]
This is with difftastic 0.23 on e.g. [this](https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/cd86181d54b5bd04705a2406e88ea900e9201111) and [this](https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/3de245ed17ad3678e2826c4af516195ed70055b1) commit. Some others are working fine.

Output from the first one with `git show --ext-diff` (middle removed because github only allows 64k characters):

<details>

```
thread 'main' panicked at 'Should have changes set in all nodes: List id:1103 content:663 {
    open_content: "",
    open_position: "244:0-0",
    children: [
        Atom id:1104 content:45 {
            content: "static",
            position: "244:0-6",
        },
        Atom id:1105 content:86 {
            content: "void",
            position: "244:7-11",
        },

[...]

    ],
    close_content: "",
    close_position: "295:1-1",
}', /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/difftastic-0.23.0/src/syntax.rs:774:32
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:498:5
   1: core::panicking::panic_fmt
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:116:14
   2: difft::syntax::change_positions_
   3: difft::diff_file
   4: difft::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
fatal: external diff died, stopping at subprojects/gst-plugins-good/gst/videocrop/gstvideocrop.c
```

**Comment by Wilfred** [comment]
Thanks for the report. This was fixed in d06f357c93873d673c9328f501347fc64f759eaf, and I've just cut a release 0.23.1 which includes this commit.

**Comment by sdroege** [comment]
Thanks!
